File: algorithms/python/56-merge-intervals.py
# https://leetcode.com/problems/merge-intervals/
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals
        intervals.sort()
        solution = []
        rnge = intervals[0]
        for i in range(1, len(intervals)):
            if (rnge[1] < intervals[i][0]):
                solution.append(rnge)
                rnge = intervals[i]
            else: ## (rnge[1] >= intervals[i][0]):
                rnge[1] = max(rnge[1], intervals[i][1])
        solution.append(rnge)
        return solution
            

File: algorithms/swift/56-merge-intervals.swift
// https://leetcode.com/problems/merge-intervals/

class Solution {
    func merge(_ intervals: [[Int]]) -> [[Int]] {
        if (intervals.count == 1) {
            return intervals
        }
        var intervals = intervals
        intervals.sort {
            $0[0] < $1[0]
        }
        
        var sol = [[Int]]()
        var range = intervals[0]
        for interval in intervals {
            if (range[1] < interval[0]) {
                sol.append(range)
                range = interval
            }
            if (range[1] >= interval[0]) {
                range[1] = max(range[1], interval[1])
            }
        }
        sol.append(range)
        return sol
    }
}
